decompiler/pipeline/dataflowanalysis/dead_loop_elimination.py

```diff
@@ -86,21 +90,21 @@ def _evaluate_branch_condition(
         """
         Evaluate Branch.condition with z3 and return (satisfiable edge, unsatisfiable edge) if one edge is unsatisfiable.
         """
-        try:
-            condition = self._logic_converter.convert(condition)
-        except ValueError as value_error:
-            warning(f"[{self.__class__.__name__}] {str(value_error)}")
-            return None
         out_edges = graph.get_out_edges(block)
         assert len(out_edges) == 2, "expect two edges for Branch"
         sat_edge, unsat_edge = None, None
         for edge in out_edges:
-            # we need one sat and one unsat edge for pruning
-            if self._is_invalid_edge(edge, condition):
-                assert not unsat_edge, "two unsat branches on first visit"
-                unsat_edge = edge
-            else:
-                sat_edge = edge
+            try:
+                condition2 = self._logic_converter.convert(condition)
```

Comment:
Unfortunately it seems that making a variable to hold the condition still doesn't stop it from being simplified, leading to some test cases to fail (either due to simplifying the wrong formula or invalid formula). Thus, it seemed that resetting the condition by converting it multiple times worked better here. I wonder if there's a better way to do this? :thinking:

Reply:
Nevermind, I managed to undo this change. However, I had to implement a useless ```make_variable()``` for Z3Converter. I'm not sure if there's a cleaner way to do this otherwise.